pkg/quarkus/v1alpha/api.go

```diff
@@ -116,3 +158,82 @@ func (p *createAPISubcommand) InjectResource(res *resource.Resource) error {
 
 	return nil
 }
+
+// findOldFilesForReplacement verifies marker (## marker) and if it found then merge new api CRD file to the odler logic
+func findOldFilesForReplacement(path, newfile string) bool {
+
+	f, err := os.Open(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// remember to close the file at the end of the program
+	defer f.Close()
+
+	// read the file line by line using scanner
+	scanner := bufio.NewScanner(f)
+	var foundMarker bool
+	for scanner.Scan() {
+		// do something with a line
+		if scanner.Text() == "## marker" {
+			foundMarker = true
+			break
+		}
+	}
+
+	if foundMarker {
+		scanner.Scan()
+		catLine := scanner.Text()
+
+		splitByPipe := strings.Split(catLine, "|")
+
+		finalString := strings.TrimSuffix(strings.TrimPrefix(strings.TrimSpace(splitByPipe[0]), "cat"), "target/kubernetes/kubernetes.yml")
+
+		updatedLine := "	" + "cat" + finalString + newfile + " target/kubernetes/kubernetes.yml" + " |" + splitByPipe[1]
+		// fmt.Printf("merge : %s\n", merge)
+
+		if err := scanner.Err(); err != nil {
+			log.Fatal(err, "Failed to scan the line from Makefile")
+			return false
+		}
+
+		// ReplaceInFile replaces all instances of old with new in the file at path.
+		err = util.ReplaceInFile(path, catLine, updatedLine)
+		if err != nil {
+			log.Fatal(err, "Failed to Replace in Makefile")
+			return false
+		}
+	}
+
+	return foundMarker
+}
+
+const (
+	makefileBundleCRDFile = `target/kubernetes/%[1]s.%[2]s-%[3]s.yml`
+)
+
+const (
+	makefileBundleVarFragment = `
+##@Bundle
+.PHONY: bundle
+bundle:  ## Generate bundle manifests and metadata, then validate generated files.
```

Comment:
I won't hold this PR up but I think we need to make this depend on a build target. 

```Makefile
.PHONY: build
build:
    mvn install

.PHONY: bundle
bundle: build
    cat target...
```

Basically in our other projects if we run `make  bundle` it will do a generation and create the manifests required. But in Java, it won't. But that's because we're not calling anything from `make bundle`.

Reply:
Yes agree, we must add `mvn install` before generating the bundle else it will fail if the User is not compiled the project before generating the bundle manifest. Let me try to add this to the existing logic.

Reply:
Quick question here, when the User tries the below command at that moment itself it will generate the `/target` directory.

`make docker-build docker-push IMG=quay.io/YOURUSER/memcached-quarkus-operator:0.0.1`

Do you still think we still need to add `mvn clean install` in the bundle structure?

Reply:
Other solution is we can add like below but it will take like time for `make bundle` as `mvn clean install` will take couple of seconds

```
bundle:  ## Generate bundle manifests and metadata, then validate generated files.
	mvn clean install
## marker
	cat target/kubernetes/%[1]s.%[2]s-%[3]s.yml target/kubernetes/kubernetes.yml | operator-sdk generate bundle -q --overwrite --version 0.1.1 --default-channel=stable --channels=stable --package=%[4]s
	operator-sdk bundle validate ./bundle
```